embed.fnc

```diff
@@ -201,32 +201,27 @@
 :	    function is not to look behind.  If a parameter is marked with this
 :	    constraint, another parameter to the function must be marked with
 :	    one of the constraints below in this list.
-:   EPTR    means that not only must this pointer parameter be non-NULL, it
-:	    points to the position one byte beyond the end of a character
-:	    string.  The called function is not to look at the byte in that
-:	    position or any higher ones.  If a parameter is marked with this
-:	    constraint, another parameter to the function must be marked with
-:	    SPTR, or MPTR (described just below).  It also is fine to have
-:	    both an SPTR parameter and an MPTR one.
 :   MPTR    means that not only must this pointer parameter be non-NULL, it
 :	    points to a position somewhere in the middle of a character string.
 :	    If a parameter is marked with this constraint, another parameter to
-:	    the function must be marked with one of SPTR, EPTR, or EPTRQ
-:	    (described just below).  It also is fine to have both an SPTR
-:	    parameter and an EPTR (or EPTRQ) one.
-:   EPTRQ   is like EPTR, but the called function is equpped to handle the case
-:	    where the input SPTR and/or MPTR are equal to this parameter; they
-:	    don't have to be strictly less than it.  If a parameter is marked
-:	    with this constraint, no parameter may be marked as EPTR.
+:	    the function must be marked with one or both of SPTR, and either of
+:	    the EPTR ones (described just below).
+:   EPTRge  means that not only must this pointer parameter be non-NULL, it
+:	    points to the final character of the string that a corresponding
+:	    SPTR began and/or MPTR points to the middle of.  The called
+:	    function is not to look at any byte beyond this one.  This means
+:	    the string may be empty.
```

Comment:
This doesn't seem right.

As I read this, for the string "ABCD" the `EPTRge` tagged pointer would point at the "D" , "it points to the final character of the string" and that "D" is included in the string to be processed.

"The called function is not to look at any byte beyond this one" implies they can deref the EPTRge tagged pointer (but not any beyond that) which indicates the same.

Maybe:
```
it points immediately past the final character of the string that ...
```